static/app/components/onboardingWizard/newSidebar.tsx

```diff
@@ -133,8 +236,69 @@ interface NewSidebarProps extends Pick<CommonSidebarProps, 'orientation' | 'coll
 }
 
 export function NewOnboardingSidebar({onClose, orientation, collapsed}: NewSidebarProps) {
+  const api = useApi();
+  const organization = useOrganization();
+  const onboardingContext = useContext(OnboardingContext);
+  const {projects} = useProjects();
   const walkthrough = isDemoWalkthrough();
   const {title, description} = getPanelDescription(walkthrough);
+  const {allTasks, basicTasks} = useOnboardingTasks(
+    organization,
+    projects,
+    onboardingContext
+  );
+
+  const markCompletionTimeout = useRef<number | undefined>();
+  const markCompletionSeenTimeout = useRef<number | undefined>();
+
+  function completionTimeout(time: number): Promise<void> {
+    window.clearTimeout(markCompletionTimeout.current);
+    return new Promise(resolve => {
+      markCompletionTimeout.current = window.setTimeout(resolve, time);
+    });
+  }
+
+  function seenTimeout(time: number): Promise<void> {
+    window.clearTimeout(markCompletionSeenTimeout.current);
+    return new Promise(resolve => {
+      markCompletionSeenTimeout.current = window.setTimeout(resolve, time);
+    });
+  }
+
+  const markTasksAsSeen = useCallback(
+    async function () {
+      const unseenTasks = allTasks
+        .filter(task => taskIsDone(task) && !task.completionSeen)
+        .map(task => task.task);
+
+      // Incrementally mark tasks as seen. This gives the card completion
+      // animations time before we move each task into the completed section.
+      for (const task of unseenTasks) {
+        await seenTimeout(COMPLETION_SEEN_TIMEOUT);
+        updateOnboardingTask(api, organization, {task, completionSeen: true});
+      }
+    },
+    [api, organization, allTasks]
+  );
+
+  const markSeenOnOpen = useCallback(
+    async function () {
+      // Add a minor delay to marking tasks complete to account for the animation
+      // opening of the sidebar panel
+      await completionTimeout(INITIAL_MARK_COMPLETE_TIMEOUT);
+      markTasksAsSeen();
+    },
+    [markTasksAsSeen]
+  );
+
+  useEffect(() => {
+    markSeenOnOpen();
+
+    return () => {
+      window.clearTimeout(markCompletionTimeout.current);
+      window.clearTimeout(markCompletionSeenTimeout.current);
+    };
+  }, [markSeenOnOpen]);
```

Comment:
[All this code was copied ](https://github.com/getsentry/sentry/blob/0c2fe3f5582870a8e4216e2d1bcf33e7541aff81/static/app/components/onboardingWizard/sidebar.tsx#L106-L162)from the old design.